ext/lz4ruby/lz4ruby.c

```diff
@@ -126,6 +126,90 @@ static VALUE lz4internal_compressHC(VALUE self, VALUE header, VALUE input, VALUE
   return compress_internal(LZ4_compressHC, header, input, in_size);
 }
 
+static VALUE lz4internal_compress_with_dict(VALUE self,
+                                            VALUE header,
+                                            VALUE input,
+                                            VALUE in_size,
+                                            VALUE dictionary,
+                                            VALUE dictSize) {
```

Comment:
I don't see why you need to pass the `dictionary` and the `dictSize` when you don't do the same for the `header` (TBH I'm not sure why the library allows you to pass `in_size` as a different parameter but maybe that's for compressing only part of a stream??)